frontend/src/scalars/ui/Chart.vue

```diff
@@ -0,0 +1,598 @@
+    <div class="visual-dl-page-charts">
+        <div class="visual-dl-chart-box" :style="computedStyle">
+        </div>
+        <div class="visual-dl-chart-actions">
+            <v-btn @click="expandArea">
+                <v-icon size="20">settings_overscan</v-icon>
+            </v-btn>
+            <v-select
+                v-if="downloadLink"
+                style="width:100px;"
+                hint="download type"
+                :items="runsItems"
+                v-model="downloadType"
+            />
+            <v-btn
+                v-if="downloadLink"
+                @click="handleDownLoad">
+                <v-icon size="20">file_download</v-icon>
+            </v-btn>
+        </div>
+    </div>
+</template>
+<script>
+import('vuetify/dist/vuetify.min.css')
+
+// TODO: Consider create Vue Components for these
+
+// // components
+// import Button from 'san-mui/Button';
+// import Icon from 'san-mui/Icon';
+// import DropDownMenu from '../../common/component/DropDownMenu';
+
+// libs
+import echarts from 'echarts';
+import axios from 'axios';
+import {isFinite, flatten, maxBy, minBy, sortBy, max, min} from 'lodash';
+import {generateJsonAndDownload} from '../../common/util/downLoadFile';
+import {quantile} from '../../common/util/index';
+import moment from 'moment';
+
+// service
+import {getPluginScalarsScalars} from '../../service';
+
+const originLinesOpacity = 0.3;
+const defaultSmoothWeight = 0.6;
+const lineWidth = 1.5;
+const minQuantile = 0.05;
+const maxQuantile = 0.95;
+// the time to refresh chart data
+const intervalTime = 15;
+
+export default {
+    components: {
+        // 'ui-dropdown-menu': DropDownMenu,
+        // 'san-button': Button,
+        // 'san-icon': Icon
+    },
+    props: ['tagInfo', 'groupNameReg', 'smoothing', 'horizontal', 'sortingMethod', 'downloadLink', 'outlier', 'runs',
+            'running', 'runsItems'],
+    computed: {
+        computedStyle() {
+            return 'height:' + this.height + 'px;'
+                + 'width:' + this.width + 'px;';
+        }
+    },
+    data() {
+        return {
+            width: 400,
+            height: 300,
+            data: [
+                {
+                    name: 'train',
+                    value: []
+                }
+            ],
+            // choose run type for download file
+            downloadType: '',
+            isExpand: false,
+            originData: []
+        };
+    },
+    watch: {
+        runsItems: function(val) {
+            this.initDownloadType();
+        },
+        originData: function(val) {
+            this.setChartData();
+            this.setChartsOutlier();
+            this.setChartHorizon();
+        },
+        smoothing: function(val) {
+            this.setChartData();
+        },
+        outlier: function(val) {
+            this.setChartsOutlier();
+        },
+        horizontal: function(val) {
+            this.setChartHorizon();
+        },
+        // runs: function(val) {
```

Comment:
Why is this commented out?

Reply:
They did not define setChartsRuns(line 102) in anywhere. But I expect they add this for future purpose, so I comment it for now.